File: src/01-easy/power-of-thor-episode-1/solution.scala
import scala.io.StdIn
object Player {
    def main(args: Array[String]): Unit = {
        val Array(lx, ly, tx0, ty0) = StdIn.readLine().split(" ").map(_.toInt)
        var tx = tx0; var ty = ty0
        while (true) {
            StdIn.readLine()
            var dir = ""
            if (ty > ly) { dir += "N"; ty -= 1 }
            else if (ty < ly) { dir += "S"; ty += 1 }
            if (tx > lx) { dir += "W"; tx -= 1 }
            else if (tx < lx) { dir += "E"; tx += 1 }
            println(dir)
        }
    }
}

File: src/01-easy/power-of-thor-episode-1/solution.kt
fun main() {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    val lx = inputs[0]; val ly = inputs[1]
    var tx = inputs[2]; var ty = inputs[3]
    while (true) {
        readLine()
        var dir = ""
        if (ty > ly) { dir += "N"; ty-- } else if (ty < ly) { dir += "S"; ty++ }
        if (tx > lx) { dir += "W"; tx-- } else if (tx < lx) { dir += "E"; tx++ }
        println(dir)
    }
}
